src/ORM/DataQuery.php

```diff
@@ -1076,14 +1085,30 @@ protected function joinManyManyRelationship(
             if ($ancestorTable !== $componentBaseTable) {
                 $ancestorTableAliased = $componentPrefix . $ancestorTable;
                 $this->query->addLeftJoin(
-                    $ancestorTable,
+                    $this->augmentTable($ancestor, $ancestorTable),
                     "{$componentIDColumn} = \"{$ancestorTableAliased}\".\"ID\"",
                     $ancestorTableAliased
                 );
             }
         }
     }
 
+    /**
+     * Use this extension point to alter the table name
+     * useful for versioning for example
+     *
+     * @param $class
+     * @param $table
+     * @return mixed
+     */
+    protected function augmentTable($class, $table)
```

Comment:
Maybe update this function name to `getJoinTableName`

Reasoning for this recommendation is here https://github.com/silverstripe/silverstripe-versioned/pull/253/files#r376188510